### PR TITLE
[SDA-7890] Add oidc and host byo oidc config resources

### PR DESCRIPTION
**Pull request by gdbranco**
New endpoints under /hosted_oidc_configs
Includes:
* List
* Add
* /id Get
* /id Delete

### CODE REVIEW

File: model/clusters_mgmt/v1/host_byo_oidc_config_resource.model [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Manages hosting of byo oidc configuration.
+resource HostConfig {
```
Comment (zgalor) [inline model/clusters_mgmt/v1/host_byo_oidc_config_resource.model:18]: Is there a single host config, or do we allow multiple configs.

The way it is defined now you will have 

```
POST /oidc/host_config --body {...}

GET /oidc/host_config 

```

So you post and get to the same endpoint (which we don't usually do). What happens if you post and there is already a host config? error/override?

We usually have a collection, and post a new member to it

Comment (gdbranco) [reply model/clusters_mgmt/v1/host_byo_oidc_config_resource.model:18]: Why do I have the get in this? I expected Add to be a post only

File: model/clusters_mgmt/v1/oidc_resource.model [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Manages hosting of byo oidc configuration.
+resource Oidc {
```
[the remark this answers is not in the document]
Comment (gdbranco) [reply model/clusters_mgmt/v1/oidc_resource.model:18]: Didn't notice this. Will fix it

File: model/clusters_mgmt/v1/host_byo_oidc_config_resource.model (hunk 1 above)
Comment (zgalor) [reply model/clusters_mgmt/v1/host_byo_oidc_config_resource.model:18]: So users can't view current config?

Comment (gdbranco) [reply model/clusters_mgmt/v1/host_byo_oidc_config_resource.model:18]: They can download it from the oidc endpoint url that will be a response. Viewing the current config was not a pre requisite and they can't do so at the moment either.

Comment (gdbranco) [reply model/clusters_mgmt/v1/host_byo_oidc_config_resource.model:18]: > Is there a single host config, or do we allow multiple configs.

The host config is created based on the private key secret arn. In theory a client only needs one, but it is able to host more than one

Comment (gdbranco) [reply model/clusters_mgmt/v1/host_byo_oidc_config_resource.model]: Please check https://github.com/openshift-online/ocm-sdk-go/pull/721/files/3e6896abac7ec571b89de8636d9b0cc014a89050#diff-976f5f720f13a78e02eb86ce309c52f04d9575f62b12967ecbbc40571dc2c97a

@zgalor

**Review by ciaranRoche — APPROVED** [review]
lgtm

**Pull request merged**